Permit function list in using syntax (#814)
hyperledger-labs/solang: docs/language.rst

        uint constant byzantium_block = 4_370_000;
    }

Binding methods to types with ``using``
---------------------------------------

> The builtin types and the any user-defined types like structs, can be extended
> with methods using the ``using`` syntax. This can be done either using libraries

Reviewer: ```suggestion
We can bind methods to builtin types and any user-defined types like structs using the ``using`` syntax. This can be done either using libraries
```
Please, note that if you decide not to accept this suggestion, the comma placed between "structs" and "can be" should be removed, because we must not split the subject and the verb with a comma.

Author: I think in technical writing, it is best to avoid "we" or "you". I've specifically avoided this in the documentation so far.

Reviewer: I've seen many technical publications using "we", including research papers. I don't think it is that bad. In addition, I've always heard that we should avoid using passive voice in English. If you want to avoid we, maybe this is a better way to write the phrase:
```suggestion
Custom methods can be bound to builtin types and any user-defined types like structs using the ``using`` syntax. This can be done either using libraries
```

Author: In papers "we" means the authors of the paper did something. Here it is unclear who "we" refers to.

Author: I've adopted your change without the "custom", thanks

Reviewer: Feel free to remove `custom`. This is just a suggestion.

or free standing functions.

``using`` with free standing functions
______________________________________

> First, declare a function with 1 or more arguments. The first argument must

Reviewer: ```suggestion
First, declare a function with one or more arguments. The first argument must
```
This is just a matter of styling.

    }

    function clear_count(User memory user) {
        user.count = 0;
    }

    using {clear_count} for User global;

Now even when ``User`` is imported, the clear_count() method can be used.

.. code-block:: javascript

    import {User} from "user.sol";

    contract c {
        function foo(User memory user) {
            user.clear_count();
        }
    }

``using`` with libraries
________________________

A library may be used for handling methods on a type. First, declare a library
with all the methods you want for a type, and then bind the library to the type
with ``using``.

.. code-block:: javascript

    struct User {
        string name;
        uint name;
    }

    library UserLibrary {
        function clear_count(User user) internal {
            user.count = 0;
        }

        function inc(User user) internal {
            user.count++;
        }

        function dec(User user) internal {
            require(user.count > 0);
            user.count--;
        }
    }

    using UserLibrary for User global;

Scope for ``using``
___________________

The ``using`` statement may be scoped in various ways:

  - Globally by adding the ``global`` keyword

>   - Per file, by omitteding the ``global`` keyword

Reviewer: ```suggestion
  - Per file, by omitting the ``global`` keyword
```

  - Per contract, by putting the ``using`` declaration in a contract definition

If the scope is per contract, then the type maybe be replaced with ``*`` and

> the type from the first argument of the function will be used.

Reviewer: Please, provide an example for this case.

Author: done
